packages/store-sdk/src/cart/Optimistic.tsx

```diff
@@ -0,0 +1,68 @@
+import PQueue from 'p-queue'
+import React, { createContext, useEffect, useMemo, useState } from 'react'
+import type { FC } from 'react'
+
+import { useContext } from '../utils/useContext'
+import { Context as CartContext } from './Cart'
+import type { ContextValue as CartContextValue } from './Cart'
+
+export type Cart = Pick<CartContextValue, 'id' | 'items'>
+
+export interface Props {
+  onValidateCart?: (cart: Cart) => Promise<Cart | null>
```

Comment:
And then type this like `Cart<T>`.

You may need to do something like `<T extends Item = Item>(cart: Cart) => Promise<Cart | null>`

Reply:
Yes, bu then line 48 will break and I'll have to cast by hand in there too